examples/tokens/ERC20.vy

```diff
@@ -1,87 +1,138 @@
+# @dev Gets the balance of the specified address.
+# @param _owner The address to query the balance of.
+# @return An uint256 representing the amount owned by the passed address.
 @public
-def transfer(_to: address, _amount: int128(uint256)) -> bool:
-
-    if self.balances[msg.sender] >= _amount and \
-       self.balances[_to] + _amount >= self.balances[_to]:
-
-        self.balances[msg.sender] -= _amount  # Subtract from the sender
-        self.balances[_to] += _amount  # Add the same to the recipient
-        log.Transfer(msg.sender, _to, convert(_amount, 'uint256'))  # log transfer event.
-
-        return True
-    else:
-        return False
+@constant
+def balanceOf(_owner : address) -> uint256:
+    return self.balances[_owner]
 
 
-# Transfer allowed tokens from a specific account to another.
+# @dev Function to check the amount of tokens that an owner allowed to a spender.
+# @param _owner The address which owns the funds.
+# @param _spender The address which will spend the funds.
+# @return An uint256 specifying the amount of tokens still available for the spender.
 @public
-def transferFrom(_from: address, _to: address, _value: int128(uint256)) -> bool:
-
-    if _value <= self.allowed[_from][msg.sender] and \
-       _value <= self.balances[_from]:
+@constant
+def allowance(_owner : address, _spender : address) -> uint256:
+    return self.allowances[_owner][_spender]
 
-        self.balances[_from] -= _value  # decrease balance of from address.
-        self.allowed[_from][msg.sender] -= _value  # decrease allowance.
-        self.balances[_to] += _value  # incease balance of to address.
-        log.Transfer(_from, _to, convert(_value, 'uint256'))  # log transfer event.
 
-        return True
-    else:
-        return False
+# @dev Transfer token for a specified address
+# @param _to The address to transfer to.
+# @param _value The amount to be transferred.
+@public
+def transfer(_to : address, _value : uint256) -> bool:
+    _sender: address = msg.sender
+    self.balances[_sender] = self.balances[_sender] - _value
+    self.balances[_to] = self.balances[_to] + _value
+    log.Transfer(_sender, _to, _value)
+    return True
 
 
-# Allow _spender to withdraw from your account, multiple times, up to the _value amount.
-# If this function is called again it overwrites the current allowance with _value.
+#  @dev Transfer tokens from one address to another.
+#       Note that while this function emits an Approval event, this is not required as per the specification,
+#       and other compliant implementations may not emit the event.
+#  @param _from address The address which you want to send tokens from
+#  @param _to address The address which you want to transfer to
+#  @param _value uint256 the amount of tokens to be transferred
 @public
-def approve(_spender: address, _amount: int128(uint256)) -> bool:
+def transferFrom(_from : address, _to : address, _value : uint256) -> bool:
```

Comment:
isn't  return False still needed, or has that changed? If the if is removed, don't we still need the assertion of balances? (same question for def transfer).

Reply:
It seems that there had been a debate about "return false" vs "throw" but now the EIP is fixed to "throw". From the [EIP](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md#transfer): 
> The function SHOULD throw if the _from account balance does not have enough tokens to spend.
[OpenZeppelin's ERC20 contract](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC20/ERC20.sol#L136) doesn't return false.

 About assertions, this subtraction reverts if the balance of `_from` account is not enough. This is checked in the test.

```
self.balances[_from] = self.balances[_from] - _value
```